packages/relay/src/lib/eth.ts

```diff
@@ -524,6 +524,12 @@ export class EthImpl implements Eth {
           return predefined.INVALID_PARAMETER(0, `Invalid 'value' field in transaction param. Value must be greater than 0`);
         }
       } else {
+        // The size limit of the encoded contract posted to the mirror node can cause contract deployment transactions to fail with a 400 response code.
+        // The contract is actually deployed on the consensus node, so the contract will work.  In these cases, we don't want to return a 
+        // CONTRTACT_REVERT error.
+        if (e.isContractReverted() && e.message !== MirrorNodeClientError.messages.INVALID_HEX && process.env.ALLOW_INVALID_HEX_RESPONSE) {
```

Comment:
This will pass so far as `process.env.ALLOW_INVALID_HEX_RESPONSE` has a value vs making sure it's true.
Also i think as it is it's operating more as a `ESTIMATE_GAS_THROWS` vs what you have it named as. If the value is false you won't throw at all.

I think the below is more appropriate.
`private readonly estimateGasThrows = process.env. ESTIMATE_GAS_THROWS ? process.env. ESTIMATE_GAS_THROWS === "true" : true;`
```suggestion
        if (estimateGasThrows && e.isContractReverted() && e.message !== MirrorNodeClientError.messages.INVALID_HEX) {
```

Reply:
This would also need a test to make sure we can toggle appropriately.

Reply:
Updated.